### ISSUE TITLE
Errors with HTML parsing nested tags in <p>

### ISSUE THREAD

**Issue by siefkenj** [prettier/prettier#12887]
This is a revival of issue #5833

Currently, Prettier throws on invalid HTML input, like ```<p><p>x</p></p>```. It would be nice if Prettier was able to format this. In particular, I think Prettier should format invalid HTML that is _valid_ XML (possibly issuing a warning).

I often help students with their code, and the first thing I do is format it so I can see it properly. Throwing on this particular type of invalid HTML seems unnecessary.

**Comment by alexander-akait** [comment]
Why do you use invalid HTML? Do you use `XHTML`? I don't think prettier should support broken HTML, we don't officially support `XHTML`

**Comment by alexander-akait** [comment]
I disagree, HTML has specific handling many tags, if you need xml support you can install xml plugin and setup it to use for formatting broken HTML, also there is good asnwer https://github.com/prettier/prettier/issues/5833#issuecomment-460750181

So sorry, we don't want to change our behaviour, you can implement own plugin if you really need it, but I recommend to fix broken HTML

Also it is not easy just `HTML rules like tag X cannot be a child of tag Y should not be enforced by Prettier.`, we use `angular-parser` under the hood, and implement parser without enforcing is very huge work, also it will have many bad side effects in formatting

**Comment by siefkenj** [comment]
Have you considered a library like [htmlparser2](https://github.com/fb55/htmlparser2)? I don't know if the AST format is the same, but it should be very close, and it is error tolerant.

Formatting for valid HTML should not change at all, so this change would be backwards compatible. I haven't looked through the html plugin code, but I suspect that the parser is the only thing that would need to be changed.

**Comment by alexander-akait** [comment]
Sorry, we don't want to support broken HTML, it is like invalid syntax for JS/HTML, if you really want it you can create custom plugin and use [htmlparser2](https://github.com/fb55/htmlparser2), also there are many things is not working good in `htmlparser2` and not enouth infromation to create good output
